src/request/sign-polygon-transaction/SignPolygonTransactionApi.js

```diff
@@ -0,0 +1,117 @@
+/* global ethers */
+/* global TopLevelApi */
+/* global PolygonRequestParserMixin */
+/* global SignPolygonTransaction */
+/* global PolygonContractABIs */
+/* global Errors */
+/* global CONFIG */
+
+class SignPolygonTransactionApi extends PolygonRequestParserMixin(TopLevelApi) { // eslint-disable-line no-unused-vars
+    /**
+     * @param {KeyguardRequest.SignPolygonTransactionRequest} request
+     * @returns {Promise<Parsed<KeyguardRequest.SignPolygonTransactionRequest>>}
+     */
+    async parseRequest(request) {
+        if (!request) {
+            throw new Errors.InvalidRequestError('request is required');
+        }
+
+        /** @type {Parsed<KeyguardRequest.SignPolygonTransactionRequest>} */
+        const parsedRequest = {};
+        parsedRequest.appName = this.parseAppName(request.appName);
+        parsedRequest.keyInfo = await this.parseKeyId(request.keyId);
+        parsedRequest.keyLabel = /** @type {string} */ (this.parseLabel(request.keyLabel, false, 'keyLabel'));
+        parsedRequest.keyPath = this.parsePolygonPath(request.keyPath, 'keyPath');
+        parsedRequest.description = this.parseOpenGsnForwardRequest(
+            request,
+            ['transfer', 'transferWithApproval', 'refund'],
+        );
+        parsedRequest.request = request.request;
+        parsedRequest.relayData = this.parseOpenGsnRelayData(request.relayData);
+        parsedRequest.senderLabel = this.parseLabel(request.senderLabel); // Used for HTLC refunds
+        parsedRequest.recipientLabel = this.parseLabel(request.recipientLabel);
+        if (request.amount !== undefined) {
+            parsedRequest.amount = this.parsePositiveInteger(request.amount, false, 'amount');
+        }
+        if (request.approval !== undefined) {
+            parsedRequest.approval = {
+                tokenNonce: this.parsePositiveInteger(
+                    request.approval.tokenNonce,
+                    true,
+                    'approval.tokenNonce',
+                ),
+            };
+        }
+
+        return parsedRequest;
+    }
+
+    /**
+     *
+     * @param {KeyguardRequest.PolygonTransactionInfo} request
+     * @param {string[]} allowedMethods
```

Comment:
```suggestion
     * @param {Array<'transfer' | 'transferWithApproval' | 'refund'>} allowedMethods
```
Only a subset of possible methods is supported by this method.